src/main/java/frc/robot/utils/components/ReversibleDigitalInput.java

```diff
@@ -0,0 +1,18 @@
+package frc.robot.utils.components;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+
+/** A digital input (ie a beambreak or limit switch) whose normal state can be reversed */
+public class ReversibleDigitalInput {
+  final boolean reversed;
+  final DigitalInput input;
+
+  public ReversibleDigitalInput(int channel, boolean reversed) {
```

Comment:
This seems kind of weird. Is there ever a reason to use this class if you don't reverse it? This is a decorator and probably smoother as such:

```java
public class InvertedDigitalInput extends DigitalInput {
   // Following the pattern from super
   public InvertedDigitalInput(int channel) {
    super(channel);
  }

  // Alternatively,  you could write it as a wrapper, though I prefer the static factory to another constructor
  public static InvertedDigitalInput from(final DigitalInput input) {
    return new InvertedDigitalInput(input.getChannel());
  }

  @Override
  public boolean get() {
    return ! super.get();
  }
}
```

Then you just wrap when you want to invert:

```java
final DigitalInput beambreak = new InvertedDigitalInput(5);
final DigitalInput otherThing = InvertedDigitalInput.from(new DigitalInput(6));
```

Honestly, if I was writing it, I'd make a static factory for the channel constructor to avoid the magic number in the first place. There's no reason to let `new` get in the way of readability:

```java
final DigitalInput yetAnotherThing = InvertedDigitalInput.fromChannel(6);
```

Then I'd make the default constructor private to force use of the more readable factory.

Reply:
this class was originally made in 22 i believe with the goal of being able to easily swap normally open/closed digital inputs when we configure the robot. agree about factories over `new`